src/uct/ib/rdmacm/rdmacm_ep.c

```diff
@@ -110,10 +164,12 @@ static UCS_CLASS_CLEANUP_FUNC(uct_rdmacm_ep_t)
     uct_worker_progress_unregister_safe(&iface->super.worker->super,
                                         &self->slow_prog_id);
 
-    /* if the destroyed ep is the active one on the iface, mark it as destroyed
-     * so that arriving events on the iface won't try to access this ep */
-    if (iface->ep == self) {
-        iface->ep = UCT_RDMACM_IFACE_BLOCKED_NO_EP;
+    /* mark this ep as destroyed so that arriving events on it won't try to
+     * use it */
+    if (self->is_on_cm_ids_list) {
+        rdmacm_cm_id = (uct_rdmacm_cm_id_t *)self->rdmacm_cm_id->cm_id->context;
+        rdmacm_cm_id->ep = UCT_RDMACM_IFACE_BLOCKED_NO_EP;
```

Comment:
can we use NULL here now instead of the constant UCT_RDMACM_IFACE_BLOCKED_NO_EP? it should be possible now that the ep pointer is on the cm_id struct and not the iface